lib/runtime-core/src/codegen.rs

```diff
@@ -298,18 +298,14 @@ impl MiddlewareChain {
     }
 
     /// Run this chain with the provided function code generator, event and module info.
-    pub(crate) fn run<E: Debug, FCG: FunctionCodeGenerator<E>>(
+    pub(crate) fn run<'a, 'b: 'a, E: Debug, FCG: FunctionCodeGenerator<E>>(
         &mut self,
         fcg: Option<&mut FCG>,
-        ev: Event,
+        mut sink: EventSink<'a, 'b>,
         module_info: &ModuleInfo,
     ) -> Result<(), String> {
-        let mut sink = EventSink {
-            buffer: SmallVec::new(),
-        };
-        sink.push(ev);
         for m in &mut self.chain {
-            let prev: SmallVec<[Event; 2]> = sink.buffer.drain().collect();
+            let prev: Vec<Event> = sink.buffer.drain(..).collect();
```

Comment:
Why is `SmallVec` replaced by `Vec` here? In most cases where no middleware made changes to the opcode stream, there should only be one Event here and an heap allocation would be saved by SmallVec.